### PR TITLE
[bigassfan] Add support for standalone lights

### PR DESCRIPTION
**Pull request by mhilbush**
Signed-off-by: Mark Hilbush <mark@hilbush.com>

- Add support for standalone lights, including ON/OFF, brightness level, and color temperature. See community discussion [here](https://community.openhab.org/t/haiku-light-with-bigassfan-binding/41226/3).

- Refactor config parameters into separate config.xml.

- Fix issues from code analysis

### CODE REVIEW

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java [hunk 1]
```diff
@@ -483,6 +461,17 @@ private boolean isLightPresent() {
         return false;
     }
 
+    private static final StringType LIGHT_COLOR = new StringType("COLOR");
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: I seem to have missed this the other time the normal order (element of least surprise) is that a class first contains all constants, then all fields, then constructor(s) and then the normal methods om some sensible order.

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java [hunk 2]
```diff
@@ -483,6 +461,17 @@ private boolean isLightPresent() {
...
+
+    private boolean isLightColor() {
+        if (fanStateMap.containsKey(CHANNEL_LIGHT_COLOR)) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: This is the same as:

```java
return fanStateMap.containsKey(CHANNEL_LIGHT_COLOR) && LIGHT_COLOR.equals(fanStateMap.get(CHANNEL_LIGHT_COLOR))
```

Note that I also inverted the equals to make it more null-safe

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java [hunk 3]
```diff
@@ -565,6 +554,8 @@ private ThingStatusDetail getDetail() {
         private final long FAN_LISTENER_DELAY = 2L;
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: Can the FanListener class get its own file?

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/utils/BigAssFanConverter.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public static PercentType levelToPercent(String level) {
+        // Light will supply brightness value in range of 0-16
+        // Convert that value to a PercentType in range 0-100, which is what Dimmer item expects
+        return new PercentType((int) Math.round(Integer.parseInt(level) * BRIGHTNESS_CONVERSION_FACTOR));
+    }
+
+    /*
+     * Convert from hue range (2200-5000) to dimmer range (0-100).
+     */
+    private static final double HUE_CONVERSION_FACTOR = 28.0;
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/utils/BigAssFanConverter.java]: Please put the constants together on top of the class

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java (hunk 3 above)
Comment (mhilbush) [reply addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: I was going to do that in a separate PR. Ok?

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/utils/BigAssFanConverter.java (hunk 4 above)
Comment (mhilbush) [reply addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/utils/BigAssFanConverter.java]: Ok.

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java (hunk 1 above)
Comment (mhilbush) [reply addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: Ok

File: addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java (hunk 2 above)
Comment (mhilbush) [reply addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java]: > Note that I also inverted the equals to make it more null-safe

Good catch. Thanks.

**Comment by mhilbush** [comment]
@martinvw Thanks for the feedback. I addressed all of your comments except one, which I would prefer to do in a separate PR rather than this one. Hope that's ok with you.

**Comment by martinvw** [comment]
Thank you too.

**Pull request merged**